### PR TITLE
added keyboard event listener so that we can resize the text area with keyboard up and down key 

### PR DESCRIPTION
**Pull request by anvesh-singh**
#### What does this PR do?
solves issue 
TextArea - resize control is not keyboard accessible #7338

#### Where should the reviewer start?
only made couple of changes to code in this file
src/js/components/TextArea/TextArea.js

#### What testing has been done on this PR?

manual testing and pre defined test are passed

#### How should this be manually tested?
you can easily run the code and check if it works or not 

#### Do Jest tests follow these best practices?

- [x] `screen` is used for querying.
- [x] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [x] `asFragment()` is used for snapshot testing.

### CODE REVIEW

**Comment by anvesh-singh** [comment]
plz merge this pr as this is part of my hacktoberfest contributions i just forgot to sign off my commits thats why its failed the testing

**Comment by anvesh-singh** [comment]
@jcfilben 
Calendar - date should be selectable by pressing 'space' #7337
it also fixes this issue

File: src/js/components/TextArea/TextArea.js [hunk 1]
```diff
@@ -30,16 +28,30 @@ const TextArea = forwardRef(
...
+          Prevent default behavior of closing drop and 
+          layer when pressing Esc
+          */
           event.stopPropagation();
           event.nativeEvent.stopImmediatePropagation();
         }}
-        onKeyDown={onKeyDown}
+        onKeyDown={(event) => {
+          heightResize(event);
+        }}
```
Comment (jcfilben) [inline src/js/components/TextArea/TextArea.js:54]: What happens if someone tries to pass a function to the `onKeyDown` prop?

File: src/js/components/TextArea/TextArea.js [hunk 2]
```diff
@@ -49,6 +61,7 @@ const TextArea = forwardRef(
           focus={focus}
           value={value}
           focusIndicator={focusIndicator}
+          style={{ height: `${height}px` }}
```
Comment (jcfilben) [inline src/js/components/TextArea/TextArea.js:64]: If I use my mouse to adjust the TextArea size and make the height larger than the original size, then press the 'down' arrow, the TextArea jumps back to a smaller size

File: src/js/components/TextArea/__tests__/TextArea-test.tsx [hunk 3]
```diff
@@ -144,7 +144,7 @@ describe('TextArea', () => {
       const textArea = component.getByPlaceholderText('item');
 
       fireEvent.keyDown(textArea, keyEvent);
-      expect(capturedEvent).toEqual(expect.objectContaining(keyEvent));
+      expect(capturedEvent).toEqual(null);
```
Comment (jcfilben) [inline src/js/components/TextArea/__tests__/TextArea-test.tsx:147]: Can we avoid changing this test, this change is a sign that we may have broken existing functionality?

**Review by jcfilben — COMMENTED** [review]
Hi @anvesh-singh, added a few comments

File: src/js/components/TextArea/TextArea.js [hunk 4]
```diff
@@ -30,16 +28,30 @@ const TextArea = forwardRef(
       value: valueProp,
     });
     const [focus, setFocus] = useState();
+    const [height, setHeight] = useState(71.6); // default value
```
[the remark this answers is not in the document]
Comment (anvesh-singh) [reply src/js/components/TextArea/TextArea.js:31]: it was the default value before i just didnt want to hinder the previous design thats why i used this value can you please tell me what fill prop are you talking about

[the remark this answers is not in the document]
Comment (jcfilben) [reply src/js/components/TextArea/TextArea.js:31]: See the `fill` prop on the TextArea docs https://v2.grommet.io/textarea#fill

**Pull request closed without merging**